### PR TITLE
temporary fixes

### PR DESCRIPTION
**Pull request by vepiteski**
I started from an old version and eventually duplicated previous changes. My actual version seems to work, but second derivatives in the new JuMP 0.12 are suspect. I still struggle to produce a small example of the trouble I get when solving iteratively a Brachistochrone model at iteration 107. This does not look as a NLPModels fault but rather comes from JuMP new AD implementation.

Once JuMP initializes to 0 the hv result in hprod, the temporary fix should be removed.

### CODE REVIEW

**Comment by vepiteski** [comment]
The tests are broken because of comparisons with AMPL using a deprecated package (AMPLReader).

File: src/NLPModels.jl [hunk 1]
```diff
@@ -275,6 +278,7 @@ end
 "Evaluate the product of the Lagrangian Hessian at `(x,y)` with the vector `v`."
 function hprod(nlp :: NLPModel, x :: Array{Float64}, y :: Array{Float64}, v :: Array{Float64})
   nlp.neval_hprod += 1
+  nlp.hv = zeros(size(x)) # JPD workaround
```
Comment (dpo) [inline src/NLPModels.jl]: I'm not sure I understand this; wasn't `nlp.hv` already initialized to the vector of zeros at line 144?

File: src/NLPModels.jl [hunk 2]
```diff
@@ -282,6 +286,7 @@ end
 "Evaluate the product of the Lagrangian Hessian at `(x,y)` with the vector `v` in place."
 function hprod!(nlp :: NLPModel, x :: Array{Float64}, y :: Array{Float64}, v :: Array{Float64}, hv :: Array{Float64})
   nlp.neval_hprod += 1
+  hv = zeros(size(x)) # JPD workaround
```
Comment (dpo) [inline src/NLPModels.jl]: This creates a new vector of zeros. Wouldn't it be enough to fill the `hv` supplied by the user with zeros?

**Comment by vepiteski** [comment]
This "fix" is no longer required since JuMP itself has been fixed and performs the initialization to 0.

**Pull request merged**